src/main/java/nextstep/qna/service/QnAService.java

```diff
@@ -2,48 +2,45 @@
 
 import nextstep.qna.CannotDeleteException;
 import nextstep.qna.NotFoundException;
-import nextstep.qna.domain.*;
+import nextstep.qna.domain.answer.Answer;
+import nextstep.qna.domain.answer.Answers;
+import nextstep.qna.domain.deleteHistory.DeleteHistory;
+import nextstep.qna.domain.question.Question;
+import nextstep.qna.domain.answer.AnswerRepository;
+import nextstep.qna.domain.question.QuestionRepository;
+import nextstep.qna.domain.deleteHistory.type.ContentType;
 import nextstep.users.domain.NsUser;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 import javax.annotation.Resource;
-import java.time.LocalDateTime;
 import java.util.ArrayList;
 import java.util.List;
 
 @Service("qnaService")
 public class QnAService {
-    @Resource(name = "questionRepository")
-    private QuestionRepository questionRepository;
-
-    @Resource(name = "answerRepository")
-    private AnswerRepository answerRepository;
-
-    @Resource(name = "deleteHistoryService")
-    private DeleteHistoryService deleteHistoryService;
-
-    @Transactional
-    public void deleteQuestion(NsUser loginUser, long questionId) throws CannotDeleteException {
-        Question question = questionRepository.findById(questionId).orElseThrow(NotFoundException::new);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
-
-        List<Answer> answers = question.getAnswers();
-        for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
-
-        List<DeleteHistory> deleteHistories = new ArrayList<>();
-        question.setDeleted(true);
-        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, question.getWriter(), LocalDateTime.now()));
-        for (Answer answer : answers) {
-            answer.setDeleted(true);
-            deleteHistories.add(new DeleteHistory(ContentType.ANSWER, answer.getId(), answer.getWriter(), LocalDateTime.now()));
-        }
-        deleteHistoryService.saveAll(deleteHistories);
-    }
+  @Resource(name = "questionRepository")
+  private QuestionRepository questionRepository;
+
+  @Resource(name = "answerRepository")
+  private AnswerRepository answerRepository;
+
+  @Resource(name = "deleteHistoryService")
+  private DeleteHistoryService deleteHistoryService;
+
+  @Transactional
+  public void deleteQuestion(NsUser loginUser, long questionId) throws CannotDeleteException {
+
+    // 조회
+    Question question = questionRepository.findById(questionId).orElseThrow(NotFoundException::new);
+    Answers answers = answerRepository.findByQuestion(questionId);
```

Comment:
```suggestion
    Question question = questionRepository.findById(questionId).orElseThrow(NotFoundException::new);
```
질문을 조회하면서 답변도 같이 조회할 수 있지 않을까요?

Reply:
네넵! 제가 question에서 answer를 삭제해서 요렇게 리팩토링하게 되었는데요!
다시 answer 살리는 방향으로 수정하겠습니다!